app/src/main/java/com/pokemonshowdown/app/BattleFieldActivity.java

```diff
@@ -400,4 +436,104 @@ private void selectItem(int position) {
+            waitingDialog.setCancelable(false);
+            waitingDialog.setMax(100);
+        }
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+            BattleFieldActivity.this.runOnUiThread(new java.lang.Runnable() {
+                public void run() {
+                    waitingDialog.show();
+                }
+            });
+        }
+
+        @Override
+        protected void onProgressUpdate(Integer... values) {
+            waitingDialog.setProgress(values[0]);
+        }
+
+        @Override
+        protected Void doInBackground(Void... voids) {
+            final HttpParams httpParams = new BasicHttpParams();
+            HttpConnectionParams.setConnectionTimeout(httpParams, 5000);
+            HttpClient client = new DefaultHttpClient(httpParams);
+            HttpGet httpget = new HttpGet(APK_LOCATION);
+            HttpResponse response = null;
+            try {
+                response = client.execute(httpget);
+                status = response.getStatusLine().getStatusCode();
+                if (status == HttpStatus.SC_OK) {
+                    HttpEntity entity = response.getEntity();
+                    InputStream fileStream = entity.getContent();
+                    File file = new File(getApplicationContext().getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "temp.apk");
+                    file.createNewFile();
+                    FileOutputStream fos = new FileOutputStream(file);
+
+                    long totalLength = entity.getContentLength();
+                    int readLength = 0;
+                    int count = 0;
+                    byte data[] = new byte[1024];
+                    while ((count = fileStream.read(data)) != -1) {
+                        readLength += count;
+                        publishProgress((int) ((readLength * 100) / totalLength));
+                        fos.write(data, 0, count);
+                    }
+
+                    fos.flush();
+                    fos.close();
+                    fileStream.close();
+                }
+            } catch (IOException e) {
+                return null;
+            }
+            return null;
+
+        }
+
+        @Override
+        protected void onPostExecute(Void aVoid) {
+            super.onPostExecute(aVoid);
+
+            BattleFieldActivity.this.runOnUiThread(new java.lang.Runnable() {
+                public void run() {
+                    waitingDialog.dismiss();
+                }
+            });
+
+            if (status == HttpStatus.SC_OK) {
+                File file = new File(getApplicationContext().getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "temp.apk");
+
+                Intent promptInstall = new Intent(Intent.ACTION_VIEW)
+                        .setDataAndType(Uri.fromFile(file),
+                                "application/vnd.android.package-archive");
+                startActivity(promptInstall);
```

Comment:
I'll figure that out and fix the rest :)
On 16 Mar 2015 00:25, "Nam Thai" <notifications@github.com> wrote:

> In app/src/main/java/com/pokemonshowdown/app/BattleFieldActivity.java
> <https://github.com/NamTThai/PokemonShowdownAndroidClient/pull/81#discussion_r26453860>
> :
>
> > +        protected void onPostExecute(Void aVoid) {
> > +            super.onPostExecute(aVoid);
> > +
> > +            BattleFieldActivity.this.runOnUiThread(new java.lang.Runnable() {
> > +                public void run() {
> > +                    waitingDialog.dismiss();
> > +                }
> > +            });
> > +
> > +            if (status == HttpStatus.SC_OK) {
> > +                File file = new File(getApplicationContext().getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "temp.apk");
> > +
> > +                Intent promptInstall = new Intent(Intent.ACTION_VIEW)
> > +                        .setDataAndType(Uri.fromFile(file),
> > +                                "application/vnd.android.package-archive");
> > +                startActivity(promptInstall);
>
> What happens when the users install new .apk file? Do they lose their
> teams? If so we need to figure out how to save their teams as well in this
> pull request
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NamTThai/PokemonShowdownAndroidClient/pull/81/files#r26453860>
> .
>